File: packages/chat-python/src/rfnry_chat_protocol/stream.py
from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, ConfigDict

from rfnry_chat_protocol.identity import AssistantIdentity

StreamTargetType = Literal["message", "reasoning"]


class StreamError(BaseModel):
    model_config = ConfigDict(extra="forbid", frozen=True)

    code: str
    message: str


class StreamStartFrame(BaseModel):
    model_config = ConfigDict(extra="forbid", frozen=True)

    event_id: str
    thread_id: str
    run_id: str
    target_type: StreamTargetType
    author: AssistantIdentity


class StreamDeltaFrame(BaseModel):
    model_config = ConfigDict(extra="forbid", frozen=True)

    event_id: str
    thread_id: str
    text: str


class StreamEndFrame(BaseModel):
    model_config = ConfigDict(extra="forbid", frozen=True)

    event_id: str
    thread_id: str
    error: StreamError | None = None

File: packages/chat-typescript/src/stream.ts
import type { Identity, IdentityWire } from './identity'

export type StreamTargetType = 'message' | 'reasoning'

export type StreamError = {
  code: string
  message: string
}

export type StreamStartFrame = {
  eventId: string
  threadId: string
  runId: string
  targetType: StreamTargetType
  author: Identity
}

export type StreamStartFrameWire = {
  event_id: string
  thread_id: string
  run_id: string
  target_type: StreamTargetType
  author: IdentityWire
}

export type StreamDeltaFrame = {
  eventId: string
  threadId: string
  text: string
}

export type StreamDeltaFrameWire = {
  event_id: string
  thread_id: string
  text: string
}

export type StreamEndFrame = {
  eventId: string
  threadId: string
  error?: StreamError | null
}

export type StreamEndFrameWire = {
  event_id: string
  thread_id: string
  error?: StreamError | null
}
